### PR TITLE
chore: add build-args for git-commit etc

### PR DESCRIPTION
**Pull request by blakepettersson**
Add the ability to specify `GIT_TAG`, `GIT_COMMIT`, `BUILD_DATE` and `GIT_TREE_STATE` as optional build-args. As well as resolving #13683 (which was caused by #12620), this has the bonus of making the `docker build` slightly more deterministic (since we now have the ability to specify the same inputs into the docker build which was hitherto computed on every `docker build`).

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

**Review by 34fathombelow — CHANGES_REQUESTED** [review]
After discussing the changes over with Blake via Zoom. I'm in favor of reverting #12620 & #12640.  

Although this PR would fix the issue with our images, end users building their own custom images would still be affected. `https://argo-cd.readthedocs.io/en/stable/operator-manual/custom_tools/#byoi-build-your-own-image`

File: .github/workflows/image-reuse.yaml [hunk 1]
```diff
@@ -128,6 +128,13 @@ jobs:
           password: ${{ secrets.docker_password }}
         if: ${{ inputs.docker_image_name && inputs.push }}
 
+      - name: Setup build args for container image
```
Comment (jsoref) [inline .github/workflows/image-reuse.yaml:131]:
```suggestion
      - name: Set up build args for container image
```

**Review by jsoref — COMMENTED** [review]
I like this PR

**Comment by crenshaw-dev** [comment]
@34fathombelow lgty?

**Review by 34fathombelow — APPROVED** [review]
LGTM!

@crenshaw-dev Can we cherry-pick this to 2.7?

**Review by crenshaw-dev — APPROVED** [review]

**Comment by blakepettersson** [comment]
Thanks guys!

**Pull request merged**